Spigot-API-Patches/0222-Expose-the-Entity-Counter-to-allow-plugins-to-use-va.patch

```diff
@@ -0,0 +1,24 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: MeFisto94 <MeFisto94@users.noreply.github.com>
+Date: Fri, 28 Aug 2020 01:41:31 +0200
+Subject: [PATCH] Expose the Entity Counter to allow plugins to use valid and
+ non-conflicting Entity Ids
+
+
+diff --git a/src/main/java/org/bukkit/UnsafeValues.java b/src/main/java/org/bukkit/UnsafeValues.java
+index d3ec5084e33dff038d54cdd2aeb703a3eb25f7a7..20cb67b7f2fcc590d4530d225d2a43898fd10117 100644
+--- a/src/main/java/org/bukkit/UnsafeValues.java
++++ b/src/main/java/org/bukkit/UnsafeValues.java
+@@ -94,5 +94,12 @@ public interface UnsafeValues {
+     byte[] serializeItem(ItemStack item);
+ 
+     ItemStack deserializeItem(byte[] data);
++
++    /**
++     * Creates and returns the next EntityId available.
++     * <p>
++     * Use this when sending custom packets, so that there are no collisions on the client or server.
++     */
++    public int createEntityId();
```

Comment:
Perhaps `allocateEntityId`?

Reply:
The problem might be that it has the feeling of being able to "free" the entityId again (but `create` is no exception there).
But I guess any formulation will suffer from that

Reply:
What is with nextEntityId() ?

Reply:
nextEntityId sounds good to me, especially as it indicates that it's somehow fetching id and doesn't have the `alloc/free` notion :)

Reply:
`nextEntityId` works for me.

Reply:
I just thought of getAndIncrementEntityId to match AtomicInteger but this is such a trivial thing that's it not worth editing for